tools/check_author.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# Copyright (c) the JPEG XL Project Authors. All rights reserved.
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+
+
+"""check_author.py: Check that a given author is listed in the AUTHORS file."""
+
+import argparse
+import fnmatch
+import os
+import re
+import sys
+
+
+def IsAuthorInFile(email, name, filename):
+  """Return whether we find the name/email in the authors filename"""
+  # Organization emails have emails listed as <*@domain.com>. This matches those
+  # patterns.
+  email_pattern_regex = re.compile(r'.*(<[^>]+>)')
+
+  with open(filename, 'r') as f:
+    for line in f:
+      line = line.strip()
+      if line.startswith('#') or not line:
+        continue
+      # Exact match for a line without an email is OK.
+      if line == name:
+        return True
+      # Exact email address match is OK, even if the name is different.
+      if fnmatch.fnmatch(line, '* <%s>' % email):
+        print(
+            "User %s <%s> matched with different name %s" % (name, email, line),
+            file=sys.stderr)
+        return True
+      # Organizations often have *@domain.com email patterns which don't match
+      # the name.
+      if '*' in line:
+        m = email_pattern_regex.match(line)
+        if m and fnmatch.fnmatch('<%s>' % email, m.group(1)):
+          print("User %s <%s> matched pattern %s" % (name, email, line),
+                file=sys.stderr)
+          return True
+  return False
+
+
+def CheckAuthor(args):
+  ret = IsAuthorInFile(
+      args.email, args.name, os.path.join(args.source_dir, 'AUTHORS'))
+  if not ret:
+    print("User %s <%s> not found, please add youself to the AUTHORS file" % (
+              args.name, args.email),
+          file=sys.stderr)
+  if not ret and not args.dry_run:
+    sys.exit(1)
```

Comment:
How about merging this with the previous branch:

```suggestion
    if not args.dry_run:
      sys.exit(1)
```

Reply:
Done.